packages/yoastseo/src/parse/build/private/adapt.js

```diff
@@ -87,5 +87,7 @@ export default function adapt( tree ) {
 		return new Heading( headingLevel, attributes, children, tree.sourceCodeLocation );
 	}
 
-	return new Node( tree.nodeName, attributes, children, tree.sourceCodeLocation );
+	const commentData = tree.nodeName === "#comment" && tree.data;
+
+	return new Node( tree.nodeName, attributes, children, tree.sourceCodeLocation, commentData );
```

Comment:
Here if a node is a comment and there is `data`, we want to save this info in the tree